src/panels/lovelace/editor/hui-element-editor.ts

```diff
@@ -117,6 +118,16 @@ export abstract class HuiElementEditor<
     });
   }
 
+  protected willUpdate(changedProps: PropertyValues): void {
+    if (changedProps.has("_config")) {
+      if (changedProps.has("key")) {
+        this._guiMode = true;
```

Comment:
When changing to a new element, it could be debated if it is better to default back to gui mode or not.

If we are in yaml mode, and we do not reset back to gui mode on change, then we will need some extra handling to force the yaml editor to update its value.